### DISCUSSION TITLE
Is it possible to run it on a port other than 8090?

### DISCUSSION THREAD

**Discussion by mrcoding-dev** [pocketbase/pocketbase#314] [Q&A] [answered]
Is it possible to run it on a port other than 8090?

**Comment by ganigeorgiev** [comment, marked as answer]
Yes, you can use the `--http` flag:
```sh
./pocketbase serve --http="127.0.0.1:3000"
```

**Reply by bgbruno** [reply]
missing this info in " help "
`./pocketbase serve --help`

in answer
```
--http string       TCP address to listen for the HTTP server
                    (if domain args are specified - default to 0.0.0.0:80, otherwise - default to 127.0.0.1:8090)
```

something like
`                     (example --http="127.0.0.1:3000" )`

**Comment by maxneo4** [comment]
As a framework it is possible change the port also?
Is common to take the port from some environment variable like azure web apps.

**Reply by ganigeorgiev** [reply]
At the moment the address and port can be changed only via the cli flags, but there are plans to make it code-configurable in the near future.

One hacky workaround would be to update `os.Args` before initializing the PocketBase app and append `--http="..."` (if it is not set already).

**Reply by maxneo4** [reply]
Thanks I believe that works, but unfortunately when I used as framework I had an issue that I couldn't fix 
pocketbase@v0.10.4\ui\embed.go:10:12: pattern all:dist: no matching files found

**Reply by ganigeorgiev** [reply]
@maxneo4 I don't think this is related to the flags. Could you provide an example how you are extending PocketBase and what Go version you are using (you could also open a new discussion with more details)?

**Reply by maxneo4** [reply]
Right, I haven't used flags yet, I'm only trying to compile and use basic main example with pocket base as framework,  I will open a new discussion about that.

**Reply by godardt** [reply]
> At the moment the address and port can be changed only via the cli flags, but there are plans to make it code-configurable in the near future.
> 
> One hacky workaround would be to update `os.Args` before initializing the PocketBase app and append `--http="..."` (if it is not set already).

Any update on this? As of 0.25.0 I think it's still not possible to change the port or IP with `pocketbase.NewWithConfig`

**Reply by ganigeorgiev** [reply]
@EKami you are commenting on a very old discussion and many things have changed since then, including the PocketBase initialization. With the recent PocketBase versions you have access to the `*http.Server` instance from within the [`OnServe` hook](https://pocketbase.io/docs/go-event-hooks/#onserve), something like:

```go
app.OnServe().BindFunc(func(e *core.ServeEvent) error {
    e.Server.Addr = "localhost:8123"

    return e.Next()
})
```

Another possible option could be to call [`apis.Serve`](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.25.0/apis#Serve) manually, for example:
```go
app := pocketbase.New()

app.Bootstrap()

err := apis.Serve(app, apis.ServeConfig{
    HttpAddr: "localhost:8123",
})

if err != nil {
    log.Fatal(err)
}
```

If you have further questions, please open a new Q&A discussion but try to elaborate a little more on your use case to understand why what you are trying to do is even needed in the first place.

**Reply by godardt** [reply]
Thank you @ganigeorgiev . That's exactly what I needed

**Comment by calumk** [comment]
Is it possible to launch with only the port, and not a single fixed ip?

Example, My server has 2 IP addresses, I need pocketbase to be avaliable on both/all IP addresses the computer has.

**Reply by ganigeorgiev** [reply]
You can try to "bind" to all local IPv4 interfaces with `0.0.0.0` or `[::1]` for IPv6, eg:

```sh
./pocketbase serve --http="0.0.0.0:8090"
```

But note that this will not work with the `--https` option when issuing a TLS certificate (_this is kindof related to PR https://github.com/pocketbase/pocketbase/pull/2795_).

Alternatively, you can try to configure it on OS level by updating your `/etc/hosts` file.
